services/media/service/src/ingest/handlers/check-finish-ingest-document-handler.ts

```diff
@@ -43,10 +43,30 @@ export class CheckFinishIngestDocumentHandler extends MediaGuardedTransactionalI
     }: TypedTransactionalMessage<CheckFinishIngestDocumentCommand>,
     ownerClient: ClientBase,
   ): Promise<void> {
+    const docId = param(ingest_document_id);
+    await sql`WITH updated AS (
```

Comment:
Checked this query on a ~4500 ingest-items document. It finishes in ~80-90ms. So, I think peformance-wise it should be acceptable.

Reply:
Can you check that there is a combined index on document_id plus status on the ingest item?

Reply:
There was non, only individual indexes. Added.